orienteer-core/src/main/java/org/orienteer/core/loader/util/JarReader.java

```diff
@@ -0,0 +1,173 @@
+package org.orienteer.core.loader.util;
+
+import com.google.common.base.Optional;
+import com.google.common.collect.Sets;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+
+/**
+ * @author Vitaliy Gonchar
+ */
+public abstract class JarReader {
```

Comment:
As discussed: probably it's better to use aether at the begining: prior to parsing and etc.
And btw: if you think that code will be better if only aether will be used - I'm OK.